### ISSUE TITLE
Using tag helpers inside sections produces StartWritingScope with a writer

### ISSUE THREAD

**Issue by pranavkm** [aspnet/Razor#267]
Might not be a bug and just a consequence of not reacting to https://github.com/aspnet/Razor/pull/261. Using a tag helper inside a section produces

```C#
StartWritingScope(__razor_template_writer);
__tagHelperExecutionContext.ExecuteChildContentAsync().Wait();
EndWritingScope();
```

`RazorPage` does not have an overload for `StartWritingScope` that accepts arguments.

**Comment by rynowak** [comment]
What's the end result of this? Compilation error?

**Comment by NTaylorMullen** [comment]
Yes compilation error. I believe this was fixed indirectly in #261, will verify.

**Comment by NTaylorMullen** [comment]
This was fixed in the MVC reaction change: https://github.com/aspnet/Mvc/pull/1795

I think @pranavkm didn't see the branch initially as all, closing as invalid.
